cpp/src/arrow/array/builder_base.h

```diff
@@ -36,6 +37,24 @@
 
 namespace arrow {
 
+namespace internal {
+
+template <class Builder, class V>
+class ArrayBuilderExtraOps {
+ public:
+  Status AppendOrNull(const std::optional<V>& value) {
```

Comment:
Can you add docstrings for these two methods?

Reply:
Done.